### PR TITLE
feat(VSlider): update to v3

### LINKED ISSUE #4746: [Bug Report] v-slider crashes the browser tab when converting to percent
### Versions and Environment
**Vuetify:** 1.1.8
**Vue:** 2.5.16
**Browsers:** Chrome 68.0.3440.75, Mozilla Firefox
**OS:** Linux

### Steps to reproduce
Drag either the right or the left thumb icon. If you drag the right one, the tab normally crashes once you hit the value 59.

### Expected Behavior
The slider should work with values which are being converted before consumption. I have just used a test variable inside the component for the reproduction, but originally the computed property was supposed to emit the value (which also doesn't work).

### Actual Behavior
The slider works fine with numbers which are not being converted via a computed property, but as soon as you multiply/divide the input and output, the slider crashes after dragging it for a bit. 

### Reproduction Link
[https://codepen.io/thomaskuhlmann/pen/PBOLqb](https://codepen.io/thomaskuhlmann/pen/PBOLqb)

### Other comments
A 'percentage' option for the slider would be nice - I guess I am not the only one who wants to use the slider to help filter a data table with percentage values.

### LINKED ISSUE #5391: [Enhancement] VSlider - Automatic numerical ticks
I am using two sliders to let the user enter feet and inches.  This is the pug code used to generate the sliders:

```
    v-subheader Feet
    v-slider(model='feet' :min=3 :max=7 ticks="always" tick-size="3" tick-labels="[3,4,5,6,7]" step=1 thumb-label='always' always-dirty)
    v-subheader Inches
    v-slider(model='inches' :min=0 :max=11 ticks="always" tick-size="3" tick-labels="[1,2,3,4,5,6,7,8,9,10,11]" step=1 thumb-label='always' always-dirty)

```
It would be nice to have an option that just read the step, min, and max parameters and placed the ticks without having to define all of the labels independently.  Perhaps just having the option to pass a boolean into the tick-labels parameter would make this easy and not require a new parameter added to the API?

```
    v-subheader Feet
    v-slider(model='feet' :min=3 :max=7 ticks="always" tick-size="3" tick-labels="true" step=1 thumb-label='always' always-dirty)
    v-subheader Inches
    v-slider(model='inches' :min=0 :max=11 ticks="always" tick-size="3" tick-labels="true" step=1 thumb-label='always' always-dirty)

```

### LINKED ISSUE #6465: [Feature Request] Slider with ticks and thumb labels
### Problem to solve
We have sliders in our app that have fixed values at tick points and ones that allow the user to choose any % value. We'd like them to look more alike and to have visual cues that show the way they work more clearly than the current sliders in Vuetify.

### Proposed solution
Show users labeled tick points.
Show the selected value in the thumb label.
Allow users to use tick points and values in-between.

### LINKED ISSUE #6615: [Feature Request] Add an ability to configure slider ticks with slots and make them clickable by some option
### Problem to solve
At first I want to say thank you for the work you do.

I'm looking at the ability to configure ticks HTML for `<v-slider>` component. For now I'm able to just pass `tick-labels` as a prop and they will be displayed in `<span>`.

Can we make them configurable with slots?
And another option I see is to pass `allow-tick-click` prop to enable an ability to move between slider ranges by clicking on ticks

### Proposed solution
1) Add a new slot to `<v-slider>` to configure the tick's content, so then we can pass there our custom HTML instead of using default `<span>`s
2) Add 1 new prop to `<v-slider>` called `allow-tick-click` that will enable an ability to move between slider ranges by clicks.

### LINKED ISSUE #6818: [Feature Request] v-slider: change mouse icon to pointer when hovering / clicking on slider icon
### Problem to solve
It would be great if the mouse cursor would change into a `pointer` when hovering over the slider icon element. An extra would be to change it into a dragging / pulling hand icon, when clicking the slider icon element.

This would make it even clearer that it is really a slider and not another element (e.g. progress bar etc.). Also this is the way sliders in many other frameworks behave.

### Proposed solution
Example in the picture below:

slider3

### LINKED ISSUE #9935: [Bug Report] V-Slider convert null to min value
### Environment
**Vuetify Version:** 2.1.5
**Vue Version:** 2.6.10
**Browsers:** Chrome 80.0.3983.2
**OS:** Windows 10

### Steps to reproduce
Set value in slider

Click X to unset the value

### Expected Behavior
Put slider in the min value and return `null || undefined`

### Actual Behavior
Put slider in the min value and return `0`

### Reproduction Link
<a href="https://codepen.io/joel_rodrigues/pen/ZEYWQZp" target="_blank">https://codepen.io/joel_rodrigues/pen/ZEYWQZp</a>

### LINKED ISSUE #10566: [Bug Report] V-Text-Field not always working with V-Slider/V-Range-Slider
### Environment
**Vuetify Version:** 2.2.10
**Vue Version:** 2.6.10
**Browsers:** Chrome 80.0.3987.100
**OS:** Windows 10

### Steps to reproduce
- Use of the V-Range-Slider Min & Max values example 
- Enter a value greater than the max data value
- Range Slider Component change the value in the text field to max data value
- Enter again a value greater than the max data value
- Range Slider Component don't change the value in the text field
- Same behavior goes for the min value

### Expected Behavior
Set the text field to max data value every time the input is greater/less than max/min data value of range slider

### Actual Behavior
The Range Slider Component don't reset the text fields every time the user typed in a value out of bounds.

### Reproduction Link
<a href="https://codepen.io/amehlen/pen/JjdYQqB" target="_blank">https://codepen.io/amehlen/pen/JjdYQqB</a>

### LINKED ISSUE #12132: [Feature Request] Reverse prop for v-slider
### Problem to solve
My application (komga.org) has a book reader, which shows a `v-slider` at the bottom to display the current position in the book, and doubles as a way to go a specific page ([screenshot](https://komga.org/assets/media/guides/webreader/reader.png)).

Books can be displayed in Right To Left direction, and in that case it would make sense to reverse the slider, going from right (page 1) to left (last page).

I am aware of the Material Design [guidelines](https://material.io/design/usability/bidirectionality.html#mirroring-elements) regarding mirroring, which say:

> Do not mirror media playback buttons and the media progress indicator as they refer to the direction of tape being played, not the direction of time.

However read progress is not really passage of time, but more related to progress:

> A volume icon with a slider at its right side should be mirrored. The slider should progress RTL, and the sound waves should emerge from the right.

### Proposed solution
I would like a `reverse` prop on `v-slider`, similar to the one on `v-progress-linear`.

### PR DESCRIPTION
**Pull request by nekosaur**
## Description

resolves #4746
resolves #5391
resolves #6465
resolves #6615
resolves #6818
resolves #9935
resolves #10566
resolves #12132

## BREAKING CHANGES
- removed `vertical`. replaced by `direction`
- renamed `step` to `step-size`
- renamed `ticks` to `show-ticks`
- added new prop `ticks` that takes array `[10, 30, 50, 100]` or object `{ 0: 'spring', 1: 'summer', 2: 'fall', 3: 'winter' }`
- removed `tick-labels`

## TODO
- cleanup
- docs
- implement readonly
- what to do about label? see #6693

## Motivation and Context

v3

## Markup:

<details>

```vue
<template>
  <v-app :theme="theme">
    <div>
      <v-btn @click="rtl = !rtl">rtl</v-btn>
      <v-btn @click="theme = theme === 'light' ? 'dark' : 'light'">theme</v-btn>
      <v-btn @click="disabled = !disabled">disabled</v-btn>
    </div>
    <v-locale-provider :rtl="rtl">
      <div class="ma-10 d-flex flex-column">
        Regular
        <v-slider
          v-model="model"
          direction="vertical"
        />
        <v-slider
          v-model="model"
          direction="horizontal"
        />

        min/max
        <v-slider
          v-model="model7"
          direction="vertical"
          min="10"
          max="25"
          thumb-label="always"
        />
        <v-slider
          v-model="model7"
          direction="horizontal"
          min="10"
          max="25"
          thumb-label="always"
        />

        Step size
        <v-slider
          v-model="model2"
          direction="vertical"
          step-size="5"
        />
        <v-slider
          v-model="model2"
          direction="horizontal"
          step-size="5"
        />

        show-ticks
        <v-slider
          v-model="model3"
          direction="vertical"
          step-size="5"
          show-ticks="always"
        />
        <v-slider
          v-model="model3"
          direction="horizontal"
          step-size="5"
          show-ticks="always"
        />

        custom-ticks
        <v-slider
          v-model="model3"
          direction="vertical"
          step-size="2"
          show-ticks="always"
          :ticks="[0, 10, 90]"
        />
        <v-slider
          v-model="model3"
          direction="horizontal"
          step-size="2"
          show-ticks="always"
          :ticks="[0, 10, 90]"
        />

        thumb-label
        <v-slider
          v-model="model4"
          direction="vertical"
          thumb-label="always"
        />
        <v-slider
          v-model="model4"
          direction="horizontal"
          thumb-label="always"
        />

        sizes
        <v-slider
          v-model="model5"
          direction="vertical"
          track-size="20"
          thumb-size="40"
          tick-size="8"
          step-size="10"
          show-ticks="always"
        />
        <v-slider
          v-model="model5"
          direction="horizontal"
          track-size="20"
          thumb-size="40"
          tick-size="8"
          step-size="10"
          show-ticks="always"
        />

        disabled
        <v-slider
          v-model="model6"
          direction="vertical"
          disabled
        />
        <v-slider
          v-model="model6"
          direction="horizontal"
          disabled
        />

        crane
        <v-slider
          direction="horizontal"
          step-size="10"
          show-ticks="always"
          rounded="0"
          color="#E30425"
        />

        tick-label slot
        <v-slider
          direction="horizontal"
          show-ticks="always"
          step-size="1"
          min="0"
          max="3"
        >
          <template #tick-label="props">
            {{ seasons[props.index] }}
          </template>
        </v-slider>

        thumb-label slot
        <v-slider
          direction="horizontal"
          thumb-label="always"
          step-size="1"
          min="0"
          max="3"
        >
          <template #thumb-label="props">
            {{ emojis[props.modelValue] }}
          </template>
        </v-slider>

      </div>

      <div class="ma-10 d-flex flex-column">
        Regular
        <v-range-slider
          v-model="model1"
          direction="vertical"
        />
        <v-range-slider
          v-model="model1"
          direction="horizontal"
        />

        min/max
        <v-range-slider
          v-model="model17"
          direction="vertical"
          min="10"
          max="25"
          thumb-label="always"
        />
        <v-range-slider
          v-model="model17"
          direction="horizontal"
          min="10"
          max="25"
          thumb-label="always"
        />

        Step size
        <v-range-slider
          v-model="model12"
          direction="vertical"
          step-size="5"
        />
        <v-range-slider
          v-model="model12"
          direction="horizontal"
          step-size="5"
        />

        Ticks
        <v-range-slider
          v-model="model13"
          direction="vertical"
          step-size="5"
          show-ticks="always"
        />
        <v-range-slider
          v-model="model13"
          direction="horizontal"
          step-size="5"
          show-ticks="always"
        />

        thumb-label
        <v-range-slider
          v-model="model14"
          direction="vertical"
          thumb-label="always"
        />
        <v-range-slider
          v-model="model14"
          direction="horizontal"
          thumb-label="always"
        />

        sizes
        <v-range-slider
          v-model="model15"
          direction="vertical"
          track-size="20"
          thumb-size="40"
          tick-size="8"
          step-size="10"
          show-ticks="always"
        />
        <v-range-slider
          v-model="model15"
          direction="horizontal"
          track-size="20"
          thumb-size="40"
          tick-size="8"
          step-size="10"
          show-ticks="always"
        />

        disabled
        <v-range-slider
          v-model="model16"
          direction="vertical"
          disabled
        />
        <v-range-slider
          v-model="model16"
          direction="horizontal"
          disabled
        />

        crane
        <v-range-slider
          direction="horizontal"
          step-size="10"
          show-ticks="always"
          rounded="0"
          color="#E30425"
        />
      </div>
    </v-locale-provider>
  </v-app>
</template>

<script>
  export default {
    name: 'Playground',
    data: () => ({
      arr: [10, 30],
      foo: null,
      test: 3,
      model: 1,
      model2: 0,
      model3: 60,
      model4: 0,
      model5: 0,
      model6: 25,
      model7: 15,
      model1: [10, 30],
      model12: [10, 30],
      model13: [10, 30],
      model14: [10, 30],
      model15: [10, 30],
      model16: [10, 30],
      model17: [15, 20],
      seasons: ['spring', 'summer', 'fall', 'winter'],
      emojis: ['😭', '🙁', '😊', '😍'],
      theme: 'light',
      rtl: false,
    }),
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Review by KaelWD — APPROVED** [review]

File: packages/vuetify/src/composables/elevation.ts [hunk 1]
```diff
@@ -25,13 +26,18 @@ export const makeElevationProps = propsFactory({
...
 }, 'elevation')
 
-export function useElevation (props: ElevationProps) {
+type ElevationData = {
+  elevationClasses: Ref<string[]>
+}
+
+export function useElevation (props: ElevationProps | Ref<number | string | undefined>): ElevationData {
   const elevationClasses = computed(() => {
+    const elevation = isRef(props) ? props.value : props.elevation
```
Comment (johnleider) [inline packages/vuetify/src/composables/elevation.ts:35]: This shouldn't be here tbh. Why not handle this within the implementing component?

```ts
const { elevationClasses } = useElevation(computed(() => {
  return { elevation: props.value || props.elevation }
})
```

File: packages/vuetify/src/composables/rounded.ts [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
     type: [Boolean, Number, String],
     default: undefined,
   },
-  tile: Boolean,
 }, 'rounded')
 
-export function useRounded (props: RoundedProps, name: string) {
+export function useRounded (props: RoundedProps | Ref<RoundedValue>, name: string): RoundedData {
   const roundedClasses = computed(() => {
+    const rounded = isRef(props) ? props.value : props.rounded
```
Comment (johnleider) [inline packages/vuetify/src/composables/rounded.ts:29]: Same thing here as above. This logic doesn't belong in these types of components.

**Review by johnleider — CHANGES_REQUESTED** [review]

File: packages/vuetify/src/composables/elevation.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/composables/elevation.ts:35]: That doesn't work without this change.

File: packages/vuetify/src/components/VInput/VInput.tsx [hunk 3]
```diff
@@ -38,6 +40,11 @@ export const makeVInputProps = propsFactory({
     default: () => ([]),
   },
   persistentHint: Boolean,
+  direction: {
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VInput/VInput.tsx:43]: Makes the stylesheet simpler, vertical has to override padding and stuff on the affix slots.

**Pull request merged**